src/App.tsx

```diff
@@ -1,20 +1,75 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
+import { MovieData } from './types/MovieData';
+import { ResponseError } from './types/ReponseError';
+
+import { getMovie } from './api';
+
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const [showError, setShowError] = useState<boolean>(false);
+
+  const isMovieData = (data: MovieData | ResponseError): data is MovieData => {
+    return (data as MovieData).Title !== undefined;
+  };
+
+  useEffect(() => {
+    if (isSubmitted && query.length) {
```

Comment:
It's not a good way to handle submitting functions by useEffect. Just create a function and call it on the submit button click.